Mathlib/LinearAlgebra/Projection.lean

```diff
@@ -424,6 +425,41 @@ theorem eq_conj_prod_map' {f : E →ₗ[R] E} (h : IsProj p f) :
   · simp only [coe_prodEquivOfIsCompl, comp_apply, coe_inr, coprod_apply, map_zero,
       coe_subtype, zero_add, map_coe_ker, prodMap_apply, zero_apply, add_zero]
 
+theorem isIdempotentElem {R V : Type*} [Semiring R] [AddCommMonoid V] [Module R V]
+    {T : V →ₗ[R] V} {U : Submodule R V} (h : IsProj U T) : IsIdempotentElem T := by
+  ext; exact h.2 _ (h.1 _)
+
+protected theorem range
+    {R V : Type*} [Semiring R] [AddCommMonoid V] [Module R V] {T : V →ₗ[R] V} {U : Submodule R V}
+    (h : IsProj U T) : range T = U := by
+  ext x
+  refine ⟨fun ⟨y, hy⟩ => ?_, fun hx => ⟨x, h.map_id x hx⟩⟩
+  rw [← hy]
+  exact h.map_mem y
+
+protected theorem top (S M : Type*) [Semiring S] [AddCommMonoid M] [Module S M] :
+    IsProj (⊤ : Submodule S M) (id (R := S)) := ⟨fun _ ↦ trivial, fun _ ↦ congrFun rfl⟩
+
+theorem _root_.LinearMap.subtype_top_comp_codRestrict_top_eq_id
+    {S M : Type*} [Semiring S] [AddCommMonoid M] [Module S M] :
+    (Submodule.subtype ⊤).comp (IsProj.top S M).codRestrict = LinearMap.id := rfl
```

Comment:
It seems that this is a special case of the theorem below? If so, is there much point in keeping it around?